### PR TITLE
Improve Networks

### PR DESCRIPTION
**Pull request by josenavas**
Various small fixes on the processing graph visualization:
- The interaction with the graph is by default disabled. A button has been added to enable/disable the interaction. (clicking on the nodes still works - it only applies to zoom and panning)
- The button to show/hide the network now has text rather than a '+' or '-' sign
- The issue reported by @antongza in which the timer gets crazy has been fixed by making sure that we destroy the objects before creating a new one.
- The issue reported by @antgonza  in which it didn't allow to add the first artifact in some situations has been fixed.

### CODE REVIEW

**Comment by antgonza** [comment]
Haven't reviewed the code, first wanted to check how the GUI moved. So far 2 issues that don't allow me to move on:
- If I select a prep and try to add a new a file from the "Processing" tab I get:
```
    Traceback (most recent call last):
      File "/Users/antoniog/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/Users/antoniog/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/Users/antoniog/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 2297, in wrapper
        return method(self, *args, **kwargs)
      File "/Users/antoniog/svn_programs/qiita/qiita_pet/handlers/study_handlers/base.py", line 100, in get
        res = study_files_get_req(self.current_user.id, study_id, pt_id, atype)
      File "/Users/antoniog/svn_programs/qiita/qiita_pet/handlers/api_proxy/studies.py", line 319, in study_files_get_req
        remaining = [f for _, f in uploaded]
    UnboundLocalError: local variable 'uploaded' referenced before assignment
[E 171206 06:45:49 web:1635] 500 GET /study/files/?study_id=2&artifact_type=BIOM&prep_template_id=3 (::1) 77.50ms
```
- In the analysis network visualization if I reload the page the network gets lost:
bug-analysis-network The error:
```
     Traceback (most recent call last):
      File "/Users/antoniog/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 1141, in _when_complete
        callback()
      File "/Users/antoniog/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 1162, in _execute_method
        self._when_complete(method(*self.path_args, **self.path_kwargs),
      File "/Users/antoniog/.virtualenvs/qiita/lib/python2.7/site-packages/tornado/web.py", line 2297, in wrapper
        return method(self, *args, **kwargs)
      File "/Users/antoniog/svn_programs/qiita/qiita_core/util.py", line 121, in wrapper
        return func(*args, **kwargs)
      File "/Users/antoniog/svn_programs/qiita/qiita_pet/handlers/analysis_handlers/base_handlers.py", line 148, in get
        analysis_id, self.current_user)
      File "/Users/antoniog/svn_programs/qiita/qiita_pet/handlers/analysis_handlers/base_handlers.py", line 137, in analyisis_graph_handler_get_request
        g, full_access, nodes=nodes, edges=edges)
      File "/Users/antoniog/svn_programs/qiita/qiita_pet/util.py", line 195, in get_network_nodes_edges
        workflow_id = n[1].processing_job_workflow.id
      File "/Users/antoniog/svn_programs/qiita/qiita_db/processing_job.py", line 1038, in processing_job_workflow
        res = qdb.sql_connection.TRN.execute_fetchindex()
      File "/Users/antoniog/svn_programs/qiita/qiita_db/sql_connection.py", line 510, in wrapper
        return func(self, *args, **kwargs)
      File "/Users/antoniog/svn_programs/qiita/qiita_db/sql_connection.py", line 811, in execute_fetchindex
        return self.execute()[idx]
      File "/Users/antoniog/svn_programs/qiita/qiita_db/sql_connection.py", line 510, in wrapper
        return func(self, *args, **kwargs)
      File "/Users/antoniog/svn_programs/qiita/qiita_db/sql_connection.py", line 762, in execute
        return self._execute()
      File "/Users/antoniog/svn_programs/qiita/qiita_db/sql_connection.py", line 709, in _execute
        self._raise_execution_error(sql, sql_args, e)
      File "/Users/antoniog/svn_programs/qiita/qiita_db/sql_connection.py", line 652, in _raise_execution_error
        "Error running SQL: %s. MSG: %s\n" % (ec_lu, error.message))
    ValueError: Error running SQL: UNDEFINED_FUNCTION. MSG: function qiita.get_processing_workflow_roots(unknown) does not exist
    LINE 2:                      FROM qiita.get_processing_workflow_root...
                                      ^
    HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

**Comment by josenavas** [comment]
Not sure about the first error - but for the second: it looks like you don't have the latest version of the DB on your system. That function was defined in the previous PR.

**Comment by antgonza** [comment]
Thanks the second one is fixed by patching the db but the first one still exists. Now, we have this error: see how with the auto-reload everything seems fine but if you reload the page the artifacts get lots:
bug-analysis-network

**Comment by josenavas** [comment]
Thanks @antgonza - the 2 issues are now solved - mind taking a look at the code?

File: qiita_pet/handlers/analysis_handlers/base_handlers.py [hunk 1]
```diff
@@ -133,9 +138,15 @@ def analyisis_graph_handler_get_request(analysis_id, user):
     for a in analysis.artifacts:
         if a.processing_parameters is None:
             g = a.descendants_with_jobs
-            nodes, edges, wf_id = get_network_nodes_edges(
+            nodes, edges, a_wf_id = get_network_nodes_edges(
                 g, full_access, nodes=nodes, edges=edges)
 
+            if wf_id is None:
```
Comment (antgonza) [inline qiita_pet/handlers/analysis_handlers/base_handlers.py]: Test for this?

File: qiita_pet/templates/study_ajax/prep_summary.html [hunk 2]
```diff
@@ -263,7 +265,8 @@
     load_information_summary();
     // If the template already have an artifact attached, load the
     // artifact graph, otherwise load the page to add a new artifact
-    processingNetwork = new Vue({el: "#processing-graph-vue"});
+    // processingNetwork = new Vue({el: "#processing-graph-vue"});
```
Comment (antgonza) [inline qiita_pet/templates/study_ajax/prep_summary.html]: rm comment?

File: qiita_pet/templates/analysis_description.html [hunk 3]
```diff
@@ -20,7 +19,8 @@
 
   $(document).ready(function(){
     // Create the processing network view object
-    processingNetwork = new Vue({el: "#analysis-graph-vue"});
+    // processingNetwork = new Vue({el: "#analysis-graph-vue"});
```
Comment (antgonza) [inline qiita_pet/templates/analysis_description.html]: I guess this one is new or missed before.

**Pull request merged**